app/src/ui/changes/changes-list.tsx

```diff
@@ -560,13 +560,33 @@ export class ChangesList extends React.Component<
       { type: 'separator' },
     ]
     if (paths.length === 1) {
+      const enabled = Path.basename(path) !== GitIgnoreFileName
       items.push({
         label: __DARWIN__
           ? 'Ignore File (Add to .gitignore)'
           : 'Ignore file (add to .gitignore)',
         action: () => this.props.onIgnoreFile(path),
-        enabled: Path.basename(path) !== GitIgnoreFileName,
+        enabled,
       })
+
+      const pathComponents = path.split(Path.sep)
+      if (pathComponents.length > 1) {
+        const submenu = pathComponents.slice(0, -1).map((_, index) => {
+          const label = '/' + pathComponents.slice(0, index + 1).join('/')
```

Comment:
```suggestion
          const label = const label = Path.join(...pathComponents.slice(0, index + 1))
```
Something I should have thought of sooner, We should probably use `Path.join` since it joins the segments together using the platform-specific separator as a delimiter, then normalizes the resulting path. I believe this does remove the first slash, but I tested on my macOS and it appears to work as expected.

Reply:
We can keep the label to be platform dependent, however the gitignore format is not dependent on the platform, which is why I felt I should go with / instead of Path.join

That being said, let me know if I should adjust the labels.

Reply:
You can see that in the gitignore docs here:
https://git-scm.com/docs/gitignore

> The slash "/" is used as the directory separator. Separators may occur at the beginning, middle or end of the .gitignore search pattern.

Reply:
Yeah. Make sense. Thanks for linking to the docs on that.